IBPSA/Fluid/HeatPumps/ModularReversible/BaseClasses/PartialReversibleRefrigerantMachine.mo

```diff
@@ -0,0 +1,653 @@
+within IBPSA.Fluid.HeatPumps.ModularReversible.BaseClasses;
+partial model PartialReversibleRefrigerantMachine
+  "Model for reversible heat pumps and chillers with a refrigerant cycle"
+  extends IBPSA.Fluid.Interfaces.PartialFourPortInterface(
+    redeclare final package Medium1 = MediumCon,
+    redeclare final package Medium2 = MediumEva,
+    final m1_flow_nominal=mCon_flow_nominal,
+    final m2_flow_nominal=mEva_flow_nominal,
+    final allowFlowReversal1=allowFlowReversalCon,
+    final allowFlowReversal2=allowFlowReversalEva,
+    final m1_flow_small=1E-4*abs(mCon_flow_nominal),
+    final m2_flow_small=1E-4*abs(mEva_flow_nominal));
+
+  //General
+  replaceable package MediumCon =
+    Modelica.Media.Interfaces.PartialMedium "Medium on condenser side"
+    annotation (Dialog(tab = "Condenser"),choicesAllMatching=true);
+  replaceable package MediumEva =
+    Modelica.Media.Interfaces.PartialMedium "Medium on evaporator side"
+    annotation (Dialog(tab = "Evaporator"),choicesAllMatching=true);
+  replaceable PartialModularRefrigerantCycle refCyc constrainedby
+    PartialModularRefrigerantCycle(final use_rev=use_rev)
+    "Model of the refrigerant cycle" annotation (Placement(transformation(
+          extent={{-18,-18},{18,18}}, rotation=90)));
+  parameter Modelica.Units.SI.HeatFlowRate QUse_flow_nominal
+    "Nominal heat flow rate at useful side of the device"
+    annotation (Dialog(group="Nominal condition"));
+
+  parameter Real y_nominal "Nominal relative compressor speed"
+    annotation (Dialog(group="Nominal condition"));
```

Comment:
For the current model approaches in IBPSA, the complexity is not necessary. However, with 3-D SDF tables such as in AixLib, specifying the compressor speed at which the nominal design capacity is specified is vital. I can set the default to 1?!

Reply:
I suggest you set the default to 1 and add an assertion (as RPM is much higher than 1) or -- preferably -- update the 3D SDF tables so that they use 1 for the speed. It may be easier to restrict what unit the tables have, rather than allowing flexibility in the tables (which are generated once only) and making the downstream use of the model which is used often more complex.